# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1257970708 18000
#      Wed Nov 11 15:18:28 2009 -0500
# Node ID 645c478aa89d38a78fb95ae3b3287378de97f3d4
# Parent  8e345f2fe4d60cc48355d130d14a18292f8f0126
load-path.cc (load_path::do_find_all_first_of): take advantage of string_vector (std::list<std:string>) constructor

diff --git a/src/ChangeLog b/src/ChangeLog
--- a/src/ChangeLog
+++ b/src/ChangeLog
@@ -1,8 +1,13 @@
+2009-11-11  John W. Eaton  <jwe@octave.org>
+
+	* load-path.cc (load_path::do_find_all_first_of): Take advantage
+	of string_vector (std::list<std:string>) constructor.
+
 2009-11-11  John W. Eaton  <jwe@octave.org>
 
 	* help.cc (do_get_help_text_from_file, Fget_help_text_from_file):
 	New functions.
 	(do_get_help_text): Pass NAME by const reference.
 	* utils.cc (find_dir_in_path): New optional argument "all".
 	* load-path.cc (load_path::do_find_matching_dirs): New function.
 	(load_path::do_find_dir): Perform match on absolute directory
diff --git a/src/load-path.cc b/src/load-path.cc
--- a/src/load-path.cc
+++ b/src/load-path.cc
@@ -1344,28 +1344,17 @@ load_path::do_find_all_first_of (const s
 	    {
 	      if (all_files[i] == rel_flist[j])
 		retlist.push_back
 		  (file_ops::concat (p->dir_name, rel_flist[j]));
 	    }
 	}
     }
 
-  size_t retsize = retlist.size ();
-
-  string_vector retval (retsize);
-
-  for (size_t i = 0; i < retsize; i++)
-    {
-      retval[i] = retlist.front ();
-
-      retlist.pop_front ();
-    }
-
-  return retval;
+  return retlist;
 }
 
 string_vector
 load_path::do_dirs (void) const
 {
   size_t len = dir_info_list.size ();
 
   string_vector retval (len);
